platform/engineering/request-for-comment/tracking-submissions-from-origin-to-destination.md

```diff
@@ -0,0 +1,385 @@
+```
+
+### Other Submissions
+
+What about submissions outside the digital forms? With the IDs being generated using the context of the submission itself, IDs can be determined by the intaking system during processing or digitization.
+
+![Form 20-0995](images/tracking-submissions-from-origin-to-destination/paper-form.png)
+
+In the above example, you can gather the information for the id from the content. The content matches the values in the example above, so you would end up with the same ID!
+
+## Risks
+
+### Types
+
+Having types be global requires any system generating an ID would require that value as context. This can be mitigated by having a globally acknowledged place of record as well as only required the system or origin to be aware of Types it is a source of origin for, however it is still an external context requirement.
+
+### Context Changes
+
+If, through the process of submission, values used in the context of the id change, the resulting id would also change.
+
+**Examples**:
+
+#### Format Changes
+
+- Veteran submits evidence as a PNG file to VA.org.
+- That evidence is sent to Lighthouse where it is converted to a PDF file and send to CMP.
+
+If the Submission ID is generated using a hash of the PNG file as context, that context would be lost in the conversion to PDF.
+
+#### Context Loss
+
+- Veteran submits data for a form that is too long to fit in the resulting form area
+- When the PDF form is generated, some of the data is cut off or moved to an alternate section
+
+If the Submission ID is generated using that full data, the context will have been changed thus changing the ID.
+
+## Alternatives
+
+### Random IDs
+
+Assign a random unique ID for each submission. Assignment is much easier at origin as a properly unique value, such as a UUID, can generated without any context needed.
+
+A downside would be that the ID would need to either be included in the submission content or be purposely carried by each system the submission travels through in the metadata of the requests.
+
+### Central ID Map
+
+Each system in the submission chain generates its own unique ID. Once a submission has moved from SystemA to SystemB, SystemA will report to a central location the corelation of SystemA.id to SystemB.id.
+A chain would be formed in the central system such that `SystemA.id -> SystemB.id -> ... -> SystemN.id`.
+
+A downside to this method is that a failure to report a link at any step would result in the loss of visibility in all of the following steps until the missing step is found. Until the link is recovered, you cannot assume the submission failed as the system could have successfully handed the submission over but failed to notify the central system.
+
+## Additions
+
+There are some additional items and thoughts.
+
+### Origin
+
+One thought is the option to add the origin to the id, either as a concatenation of an origin id and the hash or including the origin in the hash.
```

Comment:
I think that is a good idea, but maybe it doesn't need to be in the id itself?  It kind of depends if your vision is a dictionary file (DynamoDB?) or if you plan on something more complex that will allow for additional information such as origin.

Reply:
Yeah, I think the implementation will drive some of this. I recall the idea that this data would be pulled from various systems into a Power BI instance. In that case, it would be up to them how they want to store the data best.